snpe/tutorials/Inference_notebook_proposal.ipynb

```diff
@@ -0,0 +1,691 @@
+      "  (2): Conv1d(9, 9, kernel_size=(17,), stride=(1,), padding=(16,), dilation=(2,))\n",
+      "  (3): MaxPool1d(kernel_size=11, stride=11, padding=0, dilation=1, ceil_mode=False)\n",
+      "  (4): Flatten(start_dim=1, end_dim=-1)\n",
+      "  (5): LeakyReLU(negative_slope=0.01)\n",
+      "  (6): Linear(in_features=27, out_features=64, bias=True)\n",
+      "  (7): LeakyReLU(negative_slope=0.01)\n",
+      "  (8): Linear(in_features=64, out_features=32, bias=True)\n",
+      ")\n"
+     ]
+    },
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "/opt/conda/lib/python3.7/site-packages/sbi/utils/user_input_checks.py:697: UserWarning: Parameters theta has device 'cpu'. Moving theta to the data_device 'cuda:0'.Training will proceed on device 'cuda:0'.\n",
+      "  f\"Parameters theta has device '{theta.device}'. \"\n"
+     ]
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      " Neural network successfully converged after 122 epochs.\n",
+      "        -------------------------\n",
+      "        ||||| ROUND 1 STATS |||||:\n",
+      "        -------------------------\n",
+      "        Epochs trained: 122\n",
+      "        Best validation performance: -2.1384\n",
+      "        -------------------------\n",
+      "        \n"
+     ]
+    },
+    {
+     "data": {
+      "application/javascript": [
+       "\n",
+       "            setTimeout(function() {\n",
+       "                var nbb_cell_id = 5;\n",
+       "                var nbb_unformatted_code = \"infer_and_save_posterior(\\\"cuda\\\", \\\"marketplace\\\", \\\"timeseries\\\", inference_params)\";\n",
+       "                var nbb_formatted_code = \"infer_and_save_posterior(\\\"cuda\\\", \\\"marketplace\\\", \\\"timeseries\\\", inference_params)\";\n",
+       "                var nbb_cells = Jupyter.notebook.get_cells();\n",
+       "                for (var i = 0; i < nbb_cells.length; ++i) {\n",
+       "                    if (nbb_cells[i].input_prompt_number == nbb_cell_id) {\n",
+       "                        if (nbb_cells[i].get_text() == nbb_unformatted_code) {\n",
+       "                             nbb_cells[i].set_text(nbb_formatted_code);\n",
+       "                        }\n",
+       "                        break;\n",
+       "                    }\n",
+       "                }\n",
+       "            }, 500);\n",
+       "            "
+      ],
+      "text/plain": [
+       "<IPython.core.display.Javascript object>"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "infer_and_save_posterior(\"cuda\", \"marketplace\", \"timeseries\", inference_params)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "dac92177",
+   "metadata": {},
+   "source": [
+    "### (Provisional/Draft) Z. Other uncommented functions and snippets to potantially incorporate to the proper notebook"
```

Comment:
Very good catch! Yes, it will be good to include many of these, but let's first start with only the very first one in this list of extras: the `sample_posterior_with_observed`. This is sampling from the SNPE defined posterior distribution given 1 observed timeseries of reviews. So from this function we get samples of all the parameters (rho and all the others) _conditioned_ on an observed timeseries.